### PR TITLE
Fix SOCKETS_MAX_COUNT for FreeRTOS ports

### PR DESCRIPTION
**Pull request by kasper0**
## Description

- Now the SOCKETS_MAX_COUNT for FreeRTOS ports is set according to the MEMP_NUM_NETCONN provided by lwIP.
- Delete lwIP configuration for NXP MIMXRT to use the one stored in common includes for FreeRTOS.
- Adjust common lwIP configuration - based on an updated version of lwIP config for NXP MIMXRT port.

## Motivation and Context

- Resolves TODOs introduced with https://github.com/nanoframework/nf-interpreter/pull/1686/.
- Missing PLATFORM_DEPENDENT__SOCKETS_MAX_COUNT define causes a mismatch between the lwIP configuration and the CLR.
## Types of changes

- [x] Improvement (non-breaking change that improves a feature, code or algorithm)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (work on Unit Tests, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/master/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/nanoframework/nf-interpreter?pullRequest=1885) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/nanoframework/nf-interpreter?pullRequest=1885) before we can accept your contribution.<br/><br/>:x: kasper0 [sign now](https://cla.dotnetfoundation.org/nanoframework/nf-interpreter?pullRequest=1885)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/nanoframework/nf-interpreter?pullRequest=1885) it.</sub>

**Comment by josesimoes** [comment]
@kasper0 thanks for looking into this!
Please just fix that copyright header and sign the CLA and this is ready to be merged. 😃

File: targets/FreeRTOS/_common/include/lwipopts.h [hunk 1]
```diff
@@ -1,5 +1,5 @@
 //
-// Copyright (c) .NET Foundation and Contributors
+// Copyright (c) 2021 The nanoFramework project contributors
```
[the remark this answers is not in the document]
Comment (kasper0) [reply targets/FreeRTOS/_common/include/lwipopts.h]: Fixed.

**Review by josesimoes — APPROVED** [review]
LGTM!

**Pull request merged**